packages/list_bloc/lib/src/cubits/list_cubit.dart

```diff
@@ -20,5 +20,50 @@ class ListCubit<T, F> extends DataCubit<List<T>, F> {
       rethrow;
     }
   }
-// TODO(alexis): implement remove, add, replace etc
+
+  void add(T element) {
+    final data = <T>[...state.data ?? [], element];
+    emit(state.copyWith(data: data));
+  }
+
+  void remove(T element) {
+    final data = <T>[...state.data ?? []]..remove(element);
+    emit(
+      data.isEmpty
+          ? Data.empty(filter: state.filter)
+          : Data(data: data, filter: state.filter),
+    );
+  }
+
+  void removeWhere(bool Function(T) test) {
+    final data = <T>[...state.data ?? []]..removeWhere(test);
+    emit(
+      data.isEmpty
+          ? Data.empty(filter: state.filter)
+          : Data(data: data, filter: state.filter),
+    );
+  }
+
+  void _replaceFromIndexWith(int? replaceableIndex, T replaceWith) {
+    if (replaceableIndex != null && replaceableIndex >= 0) {
+      final data = <T>[...state.data ?? []]..[replaceableIndex] = replaceWith;
+      emit(Data(data: data, filter: state.filter));
+    }
+  }
+
+  void replace({
+    required T replaceWhat,
+    required T replaceWith,
+  }) {
+    final replaceableIndex = state.data?.indexOf(replaceWhat);
```

Comment:
I think we should throw an error if the item the user is trying to replace is not on the list. Otherwise, it can be hard to understand if `replace` doesn't do anything when indexOf returns -1.